src/components/Search/FrameworkDropdown.tsx

```diff
@@ -118,20 +110,15 @@ function FrameworkDropdown({ view, mode }: FrameworkDropdownProps) {
         <Select
           data-testid='dropdown-select-framework'
           label={strings.selectLabel}
-          value={frameworkName}
+          value={`${frameworkId}`}
           labelId='select-framework-label'
           className='dropdown-select'
           variant='standard'
           onChange={(e) => void handleFrameworkSelect(e)}
           name='Framework'
         >
           {Object.entries(frameworkMap).map(([id, name]) => (
-            <MenuItem
-              id={id}
-              value={name}
-              key={name}
-              className='framework-dropdown-item'
-            >
+            <MenuItem value={id} key={name} className='framework-dropdown-item'>
```

Comment:
We now use the `id` as the `value`. This triggered a series of other changes in this file. For example we need to convert back and forth between number and string because of how the types are made.